src/values/metadata_value.rs

```diff
@@ -131,14 +133,24 @@ impl<'ctx> MetadataValue<'ctx> {
     pub fn replace_all_uses_with(&self, other: &MetadataValue<'ctx>) {
         self.metadata_value.replace_all_uses_with(other.as_value_ref())
     }
-}
 
-impl AsValueRef for MetadataValue<'_> {
-    fn as_value_ref(&self) -> LLVMValueRef {
+    pub fn as_value_ref(&self) -> LLVMValueRef {
         self.metadata_value.value
     }
 }
 
+//impl AsValueRef for MetadataValue<'_> {
+    //fn as_value_ref(&self) -> LLVMValueRef {
+        //self.metadata_value.value
+    //}
+//}
+
+impl<'ctx, V: 'ctx + AsValueRef> From<V> for MetadataValue<'ctx> {
```

Comment:
If I understand correctly, this lets you turn any inkwell `AsValueRef` impl'd type into a metadata type? I don't think this is LLVM-correct.